### PR TITLE
feat: 362, 371 Added subscription support to the gateway with schema stiching

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 362, 371

### 

# Explain the feature

- Included the subscriptions support to the api gateway and changed the microservice endpoint configuration.

- Included the cors with the api gateway.

## Does this PR introduce a breaking change

Yes

Cli Generator and all the microservices end point has changed with subscription support
## Screenshots

N/A

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api-gateway-service/service.ts [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-
 /*  Creating the server based on the environment */
 const server = http.createServer( app );
 
 /* Binding the gateway with the apollo server and extracting the schema */
 ( async () => {
-  const { schema, executor } = await apiGateway.load();
+  const userService = await getRemoteSchema( {
+    uri: `http://${ process.env.USER_SERVICE_SERVICE_HOST }/graphql`,
+    subscriptionsUri: `ws://${ process.env.USER_SERVICE_SERVICE_HOST }/subscriptions`
```
Comment (deshmukhmayur) [inline packages/api-gateway-service/service.ts:23]:
```suggestion
    uri: `http://${ process.env.USER_SERVICE_SERVICE_HOST }:${ process.env.USER_SERVICE_SERVICE_PORT }/graphql`,
    subscriptionsUri: `ws://${ process.env.USER_SERVICE_SERVICE_HOST }:${ process.env.USER_SERVICE_SERVICE_PORT }/graphql`,/subscriptions`
```

Comment (riginoommen) [reply packages/api-gateway-service/service.ts]: Yes this is tested and it working

Comment (riginoommen) [reply packages/api-gateway-service/service.ts]: Included the port as you suggested

File: packages/api-gateway-service/.env.example [hunk 2]
```diff
@@ -15,8 +15,13 @@ COOKIE_PARSER=
...
+# USER_SERVICE_SERVICE_HOST="localhost"
+# HOME_SERVICE_SERVICE_HOST="localhost"
+# FEEDBACK_SERVICE_SERVICE_HOST="localhost"
+# NOTIFICATIONS_SERVICE_SERVICE_HOST="localhost"
+
+## Service Ports
+# USER_SERVICE_SERVICE_PORT="8080"
+# HOME_SERVICE_SERVICE_PORT="8081"
+# FEEDBACK_SERVICE_SERVICE_PORT="8082"
+# NOTIFICATIONS_SERVICE_SERVICE_PORT="8082"
```
Comment (deshmukhmayur) [inline packages/api-gateway-service/.env.example:27]: `8080` will be the API Gateway, so the microservices need to be on other ports.
Also, use the ports configured in the docker-compose so it's easier and compatible with docker-compose.

```suggestion
# HOME_SERVICE_SERVICE_PORT="8081"
# FEEDBACK_SERVICE_SERVICE_PORT="8082"
# USER_SERVICE_SERVICE_PORT="8083"
# NOTIFICATIONS_SERVICE_SERVICE_PORT="8084"
```

**Comment by deshmukhmayur** [comment]
@riginoommen Can you also resolve the conflicts?

**Comment by riginoommen** [comment]
> @riginoommen Can you also resolve the conflicts?

Resolved the conflicts

**Pull request merged**